### PR TITLE
Max.fxml 1166.add grouping to conv

### PR DESCRIPTION
**Pull request by maxbartel**

### CODE REVIEW

File: src/Conversion/ONNXToTOSA/Math/Conv2D.cpp [hunk 1]
```diff
@@ -85,12 +80,60 @@ class ONNXConvOpLoweringToTOSA : public OpConversionPattern<ONNXConvOp> {
...
-    Value conv2D = tosa::CreateOpAndInfer<tosa::Conv2DOp>(rewriter,
-        op->getLoc(), newConvOutputType, newInput, newWeight, bias, pads,
-        strides, dilations);
+      conv2D = tosa::CreateOpAndInfer<tosa::Conv2DOp>(rewriter, op->getLoc(),
+          newConvOutputType, newInput, newWeight, bias, pads, strides,
+          dilations);
+    } else {
+      // Set up constants outside of loop
+      const int64_t groups = group.getSInt();
+      const int64_t sizeOfSlice = weightShape[1];
```
Comment (flemairen6) [inline src/Conversion/ONNXToTOSA/Math/Conv2D.cpp:94]: Quite a general question but what happens if `group` is not a divisor of C ? (i.e for instance here, `weightShape[1] * group != C`). Should it be checked or is ONNX mlir already checking that, or is it legal?

File: test/mlir/conversion/onnx_to_tosa/Math/Conv.mlir [hunk 2]
```diff
@@ -56,4 +56,30 @@ func.func @test_onnx_conv2d_no_dilation_pad_stride(%arg0: tensor<5x3x1024x1050xf
 // CHECK: [[OUTPUT:%.+]] = "tosa.conv2d"([[TRANSINPUT]], [[TRANSKERNEL]], [[BIAS]]) {dilation = [1, 1], pad = [0, 0, 0, 0], stride = [1, 1]} : (tensor<5x1024x1050x3xf32>, tensor<2x60x64x3xf32>, tensor<2xf32>) -> tensor<5x965x987x2xf32>
 // CHECK: [[PERM3:%.+]] = "tosa.const"() {value = dense<[0, 3, 1, 2]> : tensor<4xi64>} : () -> tensor<4xi64>
 // CHECK: {{%.+}} = "tosa.transpose"([[OUTPUT]], [[PERM3]]) : (tensor<5x965x987x2xf32>, tensor<4xi64>) -> tensor<5x2x965x987xf32>
-}
\ No newline at end of file
+}
+
+
+func.func @test_onnx_conv2d_group(%arg0: tensor<5x64x1024x1024xf32>, %arg1 : tensor<12x16x45x45xf32>, %arg2: tensor<12xf32>) ->  tensor<5x12x76x76xf32> {
```
Comment (flemairen6) [inline test/mlir/conversion/onnx_to_tosa/Math/Conv.mlir:62]: While I feel like this is correct, can we have a simpler test, for instance without strides and pads, just to be sure. This is quite of a headache to be sure it's correct

File: src/Conversion/ONNXToTOSA/Math/Conv2D.cpp (hunk 1 above)
Comment (maxbartel) [reply src/Conversion/ONNXToTOSA/Math/Conv2D.cpp:94]: This did happen when I wrote the test. The verifier of the onnx conv emits an error if the shapes and groups are not compatible.

File: test/mlir/conversion/onnx_to_tosa/Math/Conv.mlir (hunk 2 above)
Comment (maxbartel) [reply test/mlir/conversion/onnx_to_tosa/Math/Conv.mlir:62]: I will add one

**Review by flemairen6 — APPROVED** [review]

**Pull request merged**